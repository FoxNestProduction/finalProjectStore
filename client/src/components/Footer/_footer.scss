@import "../../styles/mixins";
@import "../../styles/variables";

.logo__footer {
  @include flex(center, center, row);
  img {
    width: 40px;
    height: 37px;
  }

  @media (min-width: $bp-tablet) {
    img {
      width: 44px;
      height: 40px;
    }
  }

  @media (min-width: $bp-desktop) {
    img {
      width: 62px;
      height: 57px;
    }
  }

  p {
    margin-left: 10px;
    font-family: $font-poppins;
    font-weight: 600;
    font-size: 18px;
    color: $primaryDefault-color;
  
    @media (min-width: $bp-tablet) {
      font-size: 20px;
    }
  
    @media (min-width: $bp-desktop) {
      font-size: 29px;
    }
  }
  
}


.footer {
  min-width: 320px;
  position: relative;
  @include flex(center, start, column);
  padding: 55px 32px 36px 32px;
  background-color: $bgFooter-color;
  color: $textFooter-color;
  text-align: center;
  font-size: 14px;
  font-weight: 500;
  font-family: $font-inter;

  @media (min-width: $bp-tablet) {
    @include flex(center, start, column);
    padding: 69px 64px 57px 73px;
    font-size: 12px;
  }

  @media (min-width: $bp-desktop) {
    @include flex(center, start, column);
    padding: 96px 90px 80px 96px;
    font-size: 18px;
  }
}

.footer__wrapper {
  @include flex(space-between, center, column);

  @media (min-width: $bp-tablet) {
    @include flex(space-between, center, row);
    width: 100%;
  }
}

.footer__wrapper-social {
  position: absolute;
  top: 66px;
  right: 32px;
  @include flex(center, center, row);
  gap: 18px;

  @media (min-width: $bp-tablet) {
    top: 170px;
    right: 71px;
    gap: 5vw;
  }

  @media (min-width: $bp-desktop) {
    top: 241px;
    right: 96px;
    gap: 51px;

    img {
      height: 21px;
    }
  }
}

.footer__link-wrapper {
  @include flex(center, start, column);
  margin-top: 53px;
  gap: 35px;

  @media (min-width: $bp-tablet) {
    @include flex(space-between, center, row);
    margin-top: 0;
    gap: 5vw;
  }

  @media (min-width: $bp-desktop) {
    gap: 70px;
  }
}

.footer__link {
  text-transform: uppercase;

  @media (min-width: $bp-tablet) {
    text-transform: none;
  }
}

.footer__line {
  margin-top: 33px;
  height: 1px;
  width: 100%;
  background-color: $textFooter-color;

  @media (min-width: $bp-tablet) {
    margin-top: 20px;
  }

  @media (min-width: $bp-desktop) {
    margin-top: 29px;
  }
}

.footer__text {
  margin-top: 26px;
  align-self: center;

  @media (min-width: $bp-tablet) {
    margin-top: 41px;
    align-self: start;
  }

  @media (min-width: $bp-desktop) {
    margin-top: 57px;
  }
}
