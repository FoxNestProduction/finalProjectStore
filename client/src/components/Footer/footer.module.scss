@import "../../styles/mixins.scss";
@import "../../styles/variables.scss";

.footer {
  min-width: 320px;
  position: relative;
  @include flex(center, start, column);
  padding: 55px 32px 36px 32px;
  background-color: $bgFooter-color;
  color: $textFooter-color;
  text-align: center;
  font-size: 14px;
  font-weight: 500;
  font-family: $font-inter;

  @media (min-width: $bp-tablet) {
    @include flex(center, start, column);
    padding: 69px 64px 57px 73px;
    font-size: 12px;
  }

  @media (min-width: $bp-desktop) {
    @include flex(center, start, column);
    padding: 96px 90px 80px 96px;
    font-size: 18px;
  }
}

.logo {
  @include flex(center, center, row);

  p {
    margin-left: 10px;
    font-family: $font-poppins;
    font-weight: 600;
    font-size: 18px;
    color: $primaryDefault-color;

    @media (min-width: $bp-tablet) {
      font-size: 20px;
    }

    @media (min-width: $bp-desktop) {
      font-size: 29px;
    }
  }
}

.logoIcon {
  width: 40px;
  height: 37px;

  @media (min-width: $bp-tablet) {
    width: 44px;
    height: 40px;
  }

  @media (min-width: $bp-desktop) {
    width: 62px;
    height: 57px;
  }
}

.wrapper {
  @include flex(space-between, center, column);

  @media (min-width: $bp-tablet) {
    @include flex(space-between, center, row);
    width: 100%;
  }
}

.wrapperSocial {
  position: absolute;
  top: 66px;
  right: 32px;
  @include flex(center, center, row);
  gap: 18px;

  @media (min-width: $bp-tablet) {
    top: 170px;
    right: 71px;
    gap: 5vw;
  }

  @media (min-width: $bp-desktop) {
    top: 241px;
    right: 96px;
    gap: 51px;
  }
}

.socialIcon {
  height: 18px;
  width: 18px;
  fill: $primaryDefault-color;

  @media (min-width: $bp-desktop) {
    height: 21px;
    width: 21px;
    fill: #201f1f;
  }
}

.linkWrapper {
  @include flex(center, start, column);
  margin-top: 53px;
  gap: 35px;

  @media (min-width: $bp-tablet) {
    @include flex(space-between, center, row);
    margin-top: 0;
    gap: 5vw;
  }

  @media (min-width: $bp-desktop) {
    gap: 70px;
  }
}

.link {
  text-transform: uppercase;

  @media (min-width: $bp-tablet) {
    text-transform: none;
  }
}

.line {
  margin-top: 33px;
  height: 1px;
  width: 100%;
  background-color: $textFooter-color;

  @media (min-width: $bp-tablet) {
    margin-top: 20px;
  }

  @media (min-width: $bp-desktop) {
    margin-top: 29px;
  }
}

.text {
  margin-top: 26px;
  align-self: center;

  @media (min-width: $bp-tablet) {
    margin-top: 41px;
    align-self: start;
  }

  @media (min-width: $bp-desktop) {
    margin-top: 57px;
  }
}
