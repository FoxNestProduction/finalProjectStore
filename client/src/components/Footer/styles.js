export const stylesLink = {
  color: 'text.header',
  fontFamily: 'fontFamily',
  fontWeight: 'fontWeightMedium',
  alignSelf: {
    mobile: 'start',
    tablet: 'center',
  },
  textDecoration: 'none',
  textWrap: 'nowrap',
  p: 0,
  pl: {
    mobile: '10px',
    tablet: 0,
  },
  minWidth: '0px',
  fontSize: {
    mobile: 14,
    tablet: 12,
    desktop: 18,
  },
  textTransform: {
    mobile: 'uppercase',
    tablet: 'none',
  },
};

export const stylesWrap = {
  width: '100%',
  pt: {
    mobile: '55px',
    tablet: '69px',
    desktop: '96px',
  },
  pb: {
    mobile: '36px',
    tablet: '57px',
    desktop: '80px',
  },
};

export const stylesTopWrap = {
  justifyContent: 'space-between',
  width: '100%',
};

export const stylesLinkWrap = {
  gap: {
    mobile: '35px',
    tablet: '5vw',
    desktop: '50px',
  },
  alignItems: 'start',
  mt: {
    mobile: '53px',
    tablet: 0,
    desktop: 0,
  },
};

export const stylesText = {
  alignSelf: {
    mobile: 'center',
    tablet: 'start',
  },
  color: 'text.header',
  fontWeight: '500',
  fontSize: {
    mobile: '14px',
    tablet: '12px',
    desktop: '18px',
  },
  mt: {
    mobile: '26px',
    tablet: '41px',
    desktop: '57px',
  },
};

export const stylesLine = {
  mt: {
    mobile: '33px',
    tablet: '20px',
    desktop: '29px',
  },
  height: '1px',
  width: '100%',
  backgroundColor: 'text.header',
};

export const stylesSocialWrap = {
  position: 'absolute',
  top: {
    mobile: '65px',
    tablet: '170px',
    desktop: '240px',
  },
  right: '10px',
  gap: {
    mobile: '18px',
    tablet: '3vw',
    desktop: '51px',
  },
};

export const stylesSocial = {
  color: '#201F1F',
  p: '0',
};

export const stylesTwitter = {
  color: {
    mobile: '#201F1F',
    desktop: 'primary.main',
  },
  p: '0',
};

export const stylesSizeIcon = {
  fontSize: {
    mobile: '18px',
    desktop: '21px',
  },
};
