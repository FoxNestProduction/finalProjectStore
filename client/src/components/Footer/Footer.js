import React from 'react';
import { AppBar, Box, Container, IconButton, Stack, Toolbar, Typography, Link } from '@mui/material';
import InstagramIcon from '@mui/icons-material/Instagram';
import LinkedInIcon from '@mui/icons-material/LinkedIn';
import FacebookIcon from '@mui/icons-material/Facebook';
import TwitterIcon from '@mui/icons-material/Twitter';
import { NavLink } from 'react-router-dom';
import { stylesLink, stylesWrap, stylesTopWrap, stylesLinkWrap, stylesText, stylesLine, stylesSocial, stylesSocialWrap, stylesSizeIcon } from './styles';
import Logo from '../Logo/Logo';

const Footer = () => {
  return (
    <AppBar
      elevation={0}
      component="footer"
      position="relative"
      sx={{
        backgroundColor: 'background.footer',
        minWidth: '260px',
      }}
    >
      <Container direction="column">
        <Toolbar sx={{ p: 0 }}>
          <Stack direction="column" sx={stylesWrap}>
            <Stack
              sx={stylesTopWrap}
              direction={{
                mobile: 'column',
                tablet: 'row',
              }}
            >
              <Link component={NavLink} to="/" sx={{ textDecoration: 'none' }}>
                <Logo type="footer" />
              </Link>
              <Stack
                direction={{
                  mobile: 'column',
                  tablet: 'row',
                }}
                sx={stylesLinkWrap}
              >
                <Link component={NavLink} to="/Reviews" sx={stylesLink}>
                  Reviews
                </Link>
                <Link component={NavLink} to="/Pricing" sx={stylesLink}>
                  Pricing
                </Link>
                <Link component={NavLink} to="/AboutUs" sx={stylesLink}>
                  About Us
                </Link>
                <Link component={NavLink} to="/Contact" sx={stylesLink}>
                  Contact
                </Link>
              </Stack>
            </Stack>
            <Box sx={stylesLine} />
            <Typography component="p" sx={stylesText}>
              Â© 2023 EATLY All Rights Reserved.
            </Typography>
            <Stack direction="row" sx={stylesSocialWrap}>
              <IconButton href="https://www.instagram.com/" sx={stylesSocial}>
                <InstagramIcon sx={stylesSizeIcon} />
              </IconButton>
              <IconButton href="https://www.linkedin.com/" sx={stylesSocial}>
                <LinkedInIcon sx={stylesSizeIcon} />
              </IconButton>
              <IconButton href="https://uk-ua.facebook.com/" sx={stylesSocial}>
                <FacebookIcon sx={stylesSizeIcon} />
              </IconButton>
              <IconButton href="https://twitter.com/" sx={stylesSocial}>
                <TwitterIcon sx={stylesSizeIcon} />
              </IconButton>
            </Stack>
          </Stack>
        </Toolbar>
      </Container>
    </AppBar>
  );
};

export default Footer;
