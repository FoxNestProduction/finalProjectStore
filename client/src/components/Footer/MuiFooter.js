import React from 'react';
import {
  AppBar,
  Box,
  Button,
  Container,
  IconButton,
  Stack,
  Toolbar,
  Typography,
} from '@mui/material';
import { stylesLink, stylesWrap, stylesTopWrap, stylesLinkWrap, stylesText, stylesLine } from './styles';
import MuiLogo from '../Logo/MuiLogo';
// import styles from './footer.module.scss';

const MuiFooter = () => {
  return (
    <Container direction="column">
      <AppBar
        component="footer"
        position="static"
        sx={{
          backgroundColor: 'background.footer',
          minWidth: '320px',
        }}
      >
        <Toolbar>
          <Stack sx={stylesWrap}>
            <Stack sx={stylesTopWrap}>
              <MuiLogo />
              <Stack sx={stylesLinkWrap}>
                <Button sx={stylesLink}>Blog</Button>
                <Button sx={stylesLink}>Pricing</Button>
                <Button sx={stylesLink}>About Us</Button>
                <Button sx={stylesLink}>Contact</Button>
              </Stack>
            </Stack>
            <Box sx={stylesLine} />
            {/* <div className={styles.lineMui}> </div> */}
            <Typography sx={stylesText} component="p">
              Â© 2023 EATLY All Rights Reserved.
            </Typography>
          </Stack>
        </Toolbar>
      </AppBar>
    </Container>
  );
};

export default MuiFooter;
