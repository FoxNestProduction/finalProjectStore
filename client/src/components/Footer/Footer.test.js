import { render } from '@testing-library/react';
import Footer from '../Footer/Footer';
import { MemoryRouter } from 'react-router';

const Component = (props) => {
  return (
    <MemoryRouter>
      <Footer />
    </MemoryRouter>
  );
};

describe('Snapshot test', () => {
  test('should render Footer', () => {
    const { asFragment } = render(<Component />);

    expect(asFragment()).toMatchSnapshot();
  });
});
