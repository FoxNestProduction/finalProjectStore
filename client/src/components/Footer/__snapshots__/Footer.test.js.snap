// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test should render Footer 1`] = `
<DocumentFragment>
  <footer
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative css-1cxxkmv-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      direction="column"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1fpo0vu-MuiToolbar-root"
      >
        <div
          class="MuiStack-root css-1x9ldac-MuiStack-root"
        >
          <div
            class="MuiStack-root css-1cb81n0-MuiStack-root"
          >
            <a
              aria-current="page"
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yvn3nr-MuiTypography-root-MuiLink-root active"
              href="/"
            >
              <div
                class="MuiStack-root css-1d9cypr-MuiStack-root"
              >
                <div
                  aria-label="logo"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-wnykid-MuiButtonBase-root-MuiIconButton-root"
                  role="button"
                  tabindex="0"
                >
                  <svg>
                    logo.svg
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-10ffwrh-MuiTypography-root"
                >
                  eatly
                </p>
              </div>
            </a>
            <div
              class="MuiStack-root css-17alcl-MuiStack-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-zb2cou-MuiTypography-root-MuiLink-root"
                href="/reviews"
              >
                Reviews
              </a>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-zb2cou-MuiTypography-root-MuiLink-root"
                href="/Restaurants"
              >
                Restaurants
              </a>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-zb2cou-MuiTypography-root-MuiLink-root"
                href="/about-us"
              >
                About Us
              </a>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-zb2cou-MuiTypography-root-MuiLink-root"
                href="/contact"
              >
                Contact
              </a>
            </div>
          </div>
          <div
            class="MuiBox-root css-gzteay"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 css-809q2u-MuiTypography-root"
          >
            Â© 2023 EATLY All Rights Reserved.
          </p>
          <div
            class="MuiStack-root css-wg85rh-MuiStack-root"
          >
            <a
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1hckuxn-MuiButtonBase-root-MuiIconButton-root"
              href="https://www.instagram.com/"
              tabindex="0"
              target="blanc"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1xtcyno-MuiSvgIcon-root"
                data-testid="InstagramIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <a
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1hckuxn-MuiButtonBase-root-MuiIconButton-root"
              href="https://www.linkedin.com/"
              tabindex="0"
              target="blanc"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1xtcyno-MuiSvgIcon-root"
                data-testid="LinkedInIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <a
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1hckuxn-MuiButtonBase-root-MuiIconButton-root"
              href="https://uk-ua.facebook.com/"
              tabindex="0"
              target="blanc"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1xtcyno-MuiSvgIcon-root"
                data-testid="FacebookIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <a
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1hckuxn-MuiButtonBase-root-MuiIconButton-root"
              href="https://twitter.com/"
              tabindex="0"
              target="blanc"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1xtcyno-MuiSvgIcon-root"
                data-testid="TwitterIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
